File: CentroHipicoAPI/CentroHipicoAPI.Datos/Modelos/CentroHipico/Carrera.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace CentroHipicoAPI.Datos.Modelos.CentroHipico
{
    public partial class Carrera
    {
        public Carrera()
        {
            CarreraDetalles = new HashSet<CarreraDetalle>();
            CarreraEjemplares = new HashSet<CarreraEjemplar>();
        }

        public int Id { get; set; }
        public int NumeroCarrera { get; set; }
        public DateTime FechaCarrera { get; set; }
        public string Ubicacion { get; set; }
        public bool EstaActiva { get; set; }
        public bool EstaCerrada { get; set; }
        public int? Ganador { get; set; }
        public decimal? MontoGanancia { get; set; }
        public decimal? MontoSubTotal { get; set; }
        public decimal? MontoTotal { get; set; }
        public DateTime? FechaCreacion { get; set; }
        public DateTime? FechaActualizacion { get; set; }

        public virtual ICollection<CarreraDetalle> CarreraDetalles { get; set; }
        public virtual ICollection<CarreraEjemplar> CarreraEjemplares { get; set; }
    }
}

File: CentroHipicoApp/src/app/core/models/carrera.ts
import { Cliente } from "./cliente";
import { Ejemplar } from "./ejemplar";

export interface Carrera {
    id: number;
    fechaCarrera: Date;
    estaActiva: boolean;
    estaCerrada: boolean;
    ganador: number | null;
    montoGanancia: number | null;
    montoSubTotal: number | null;
    montoTotal: number | null;
    ubicacion: string;
}

export interface CarreraResponse {
    carrera: Carrera;
    ejemplares: CarreraEjemplar[];
    detalles: CarreraDetalleResponse[];
}

export interface CarreraEjemplar {
    id: number;
    idCarrera: number;
    ejemplar: Ejemplar;
    esFavorito: boolean;

}

export interface CarreraDetalle {
    id: number;
    idCarrera: number;
    idCliente: number;
    idEjemplar: number;
    montoApuesta: number;
}

export interface CarreraDetalleResponse {
    id: number;
    idCarrera: number;
    cliente?: Cliente;
    ejemplar: CarreraEjemplar;
    montoApuesta?: number;
}

export interface CarreraGanador {
    idCarrera: number;
    idCliente: number;
}



